src/simtools/production_configuration/limits_calculation.py

```diff
@@ -0,0 +1,161 @@
+        bin_edges : np.ndarray
+            Array of bin edges.
+        loss_fraction : float
+            Fraction of events to be lost.
+        axis : int, optional
+            Axis along which to sum the event weights. Default is 0.
+        limit_type : str, optional
+            Type of limit ('lower' or 'upper'). Default is 'lower'.
+
+        Returns
+        -------
+        int
+            Bin index where the threshold is reached.
+        float
+            Bin edge value corresponding to the threshold.
+        """
+        projection = np.sum(event_weight_array, axis=axis)
+        bin_edge_value = None
+        cumulative_sum = None
+        if limit_type == "upper":
+            cumulative_sum = np.cumsum(projection)
+
+        elif limit_type == "lower":
+            cumulative_sum = np.cumsum(projection[::-1])
+
+        total_events = np.sum(projection)
+        threshold = (1 - loss_fraction) * total_events
+        bin_index = np.searchsorted(cumulative_sum, threshold)
+        if limit_type == "upper":
+            bin_edge_value = bin_edges[bin_index]
+        elif limit_type == "lower":
+            bin_edge_value = bin_edges[-bin_index]
+        return bin_index, bin_edge_value
+
+    def compute_lower_energy_limit(self, loss_fraction):
+        """
+        Compute the lower energy limit based on the event loss fraction.
+
+        Parameters
+        ----------
+        loss_fraction : float
+            Fraction of events to be lost.
+
+        Returns
+        -------
+        float
+            Lower energy limit.
+        """
+        event_weight_array = np.column_stack(
+            [
+                self.event_weight__ra3d__log10_e__[name]
+                for name in self.event_weight__ra3d__log10_e__.dtype.names
+            ]
+        )
+        bin_edges = self.event_weight__ra3d__log10_e__.meta["y_bin_edges"]
+        _, lower_bin_edge_value = self.compute_threshold(
+            event_weight_array, bin_edges, loss_fraction, axis=0, limit_type="lower"
+        )
+        return 10**lower_bin_edge_value
+
+    def compute_upper_radial_distance(self, loss_fraction):
+        """
+        Compute the upper radial distance based on the event loss fraction.
+
+        Parameters
+        ----------
+        loss_fraction : float
+            Fraction of events to be lost.
+
+        Returns
+        -------
+        float
+            Upper radial distance.
+        """
+        event_weight_array = np.column_stack(
+            [
+                self.event_weight__ra3d__log10_e__[name]
+                for name in self.event_weight__ra3d__log10_e__.dtype.names
+            ]
+        )
+        bin_edges = self.event_weight__ra3d__log10_e__.meta["x_bin_edges"]
+        _, upper_bin_edge_value = self.compute_threshold(
+            event_weight_array, bin_edges, loss_fraction, axis=1, limit_type="upper"
+        )
+        return upper_bin_edge_value
+
+    def compute_viewcone(self, loss_fraction):
```

Comment:
Assume this is in 'deg'? Do we want to use units?

Reply:
Not sure its needed here: There are no units in the tables, we don't write them out currently and there are no calculations involved, but only extracting the bin values (degree, TeV). I have added some comments about the units and to the print outs - let me know if you think this is sufficient.